### PR TITLE
Add mate_session_check_accelerated helpers

### PR DESCRIPTION
**Pull request by rbuj**
tools/mate-session-check-accelerated.{c,h}
tools/mate-session-check-accelerated-gles-helper.c
tools/mate-session-check-accelerated-gl-helper.c

based on https://github.com/GNOME/gnome-session/commit/3aafcf0f0129e28b5c52f32284f1164a93867ea3

configure.ac
tools/Makefile.am based on:

based on https://github.com/GNOME/gnome-session/commit/eeefdc8e1a436d0c58cea756e8a3ea63e59145b5

### CODE REVIEW

**Comment by sc0w** [comment]
`cppcheck` says:

`[tools/mate-session-check-accelerated-gles-helper.c:190]: (style) Condition 'renderer!=NULL' is always false`

This change fixes the warning with cppcheck:
```patch
diff --git a/tools/mate-session-check-accelerated-gles-helper.c b/tools/mate-session-check-accelerated-gles-helper.c
index ff6a62e..c802daa 100644
--- a/tools/mate-session-check-accelerated-gles-helper.c
+++ b/tools/mate-session-check-accelerated-gles-helper.c
@@ -166,7 +166,9 @@ int
 main (int argc,
       char **argv)
 {
+#ifdef GDK_WINDOWING_X11
         char *renderer = NULL;
+#endif
         GOptionContext *context;
         int ret = HELPER_NO_ACCEL;
         GError *error = NULL;
@@ -185,7 +187,6 @@ main (int argc,
 
 #ifdef GDK_WINDOWING_X11
         renderer = get_gles_renderer ();
-#endif
 
         if (renderer != NULL) {
                 if (print_renderer)
@@ -195,6 +196,7 @@ main (int argc,
                 else
                         ret = HELPER_ACCEL;
         }
+#endif
 
 out:
         return ret;

**Comment by rbuj** [comment]
There is also a [patch](https://src.fedoraproject.org/rpms/gnome-session/blob/master/f/0001-check-accelerated-gles-Use-eglGetPlatformDisplay-EXT.patch) for Fedora.

**Comment by raveit65** [comment]
No need to change this in travis.yml again.
But mesa-libGLES-devel pulls in mesa-libGL-devel in fedora because of dependencies.
Same for 
libXau-devel
libXcomposite-devel

from mock build log:
```
Dependencies resolved.
=================================================================================================================
 Package                          Architecture  Version                             Repository              Size
=================================================================================================================
Installing:
 dbus-glib-devel                  x86_64        0.110-3.fc29                        fedora                  69 k
 desktop-file-utils               x86_64        0.23-10.fc30                        fedora                  65 k
 gtk3-devel                       x86_64        3.24.8-1.fc30                       fedora                 4.5 M
 libSM-devel                      x86_64        1.2.3-2.fc30                        fedora                  12 k
 libXtst-devel                    x86_64        1.2.3-9.fc30                        fedora                  16 k
 mate-common                      noarch        1.22.0-1.fc30                       fedora                  20 k
 pangox-compat-devel              x86_64        0.0.2-14.fc30                       fedora                  10 k
 xmlto                            x86_64        0.0.28-11.fc30                      fedora                  49 k
 xorg-x11-xtrans-devel            noarch        1.3.5-8.fc30                        fedora                  82 k
 mesa-libGLES-devel               x86_64        19.0.2-3.fc30                       updates-testing         60 k
 systemd-devel                    x86_64        241-7.gita2eaa1c.fc30               updates-testing        344 k
Installing dependencies:

<cut>
 libXau                           x86_64        1.0.9-1.fc30                        fedora                  30 k
 libXau-devel                     x86_64        1.0.9-1.fc30                        fedora                  14 k
 libXcomposite                    x86_64        0.4.4-16.fc30                       fedora                  22 k
 libXcomposite-devel              x86_64        0.4.4-16.fc30                       fedora                  16 k
<cut>
 mesa-libEGL                      x86_64        19.0.2-3.fc30                       updates-testing        108 k
 mesa-libEGL-devel                x86_64        19.0.2-3.fc30                       updates-testing         33 k
 mesa-libGL                       x86_64        19.0.2-3.fc30                       updates-testing        150 k
 mesa-libGL-devel                 x86_64        19.0.2-3.fc30                       updates-testing        163 k
 mesa-libGLES                     x86_64        19.0.2-3.fc30                       updates-testing         16 k
<cut>
```
For official fedora builds is use only this build requires:
BuildRequires:  dbus-glib-devel
BuildRequires:  desktop-file-utils
BuildRequires:  gtk3-devel
BuildRequires:  mesa-libGLES-devel
BuildRequires:  libSM-devel
BuildRequires:  mate-common
BuildRequires:  pangox-compat-devel
BuildRequires:  systemd-devel
BuildRequires:  xmlto
BuildRequires:  libXtst-devel
BuildRequires:  xorg-x11-xtrans-devel

File: tools/mate-session-check-accelerated-gles-helper.c [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                g_error ("Can't parse command line: %s\n", error->message);
+                g_error_free (error);
+                goto out;
+        }
+
+#ifdef GDK_WINDOWING_X11
+        renderer = get_gles_renderer ();
+#endif
+
+        if (renderer != NULL) {
```
Comment (sc0w) [inline tools/mate-session-check-accelerated-gles-helper.c]: please fix cppcheck warning:

`[tools/mate-session-check-accelerated-gles-helper.c:190]: (style) Condition 'renderer!=NULL' is always false`

if `GDK_WINDOWING_X11` is false, the condition `if (renderer != NULL)` is useless code

Comment (rbuj) [reply tools/mate-session-check-accelerated-gles-helper.c]: I think that is fixed now.

**Comment by rbuj** [comment]
@raveit65 
done

File: tools/mate-session-check-accelerated-gles-helper.c (hunk 1 above)
Comment (sc0w) [reply tools/mate-session-check-accelerated-gles-helper.c]: yes, thanks!

**Comment by sc0w** [comment]
deprecated functions detected in this PR:

```
mate-session-check-accelerated-gles-helper.c:71:9: warning: ‘gdk_error_trap_push’ is deprecated: Use 'gdk_x11_display_error_trap_push' instead [-Wdeprecated-declarations]
         gdk_error_trap_push ();
         ^~~~~~~~~~~~~~~~~~~

mate-session-check-accelerated-gles-helper.c:153:9: warning: ‘gdk_error_trap_pop_ignored’ is deprecated: Use 'gdk_x11_display_error_trap_pop_ignored' instead [-Wdeprecated-declarations]
         gdk_error_trap_pop_ignored ();
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

and mate-session-manager has no deprecated functions in master, so, I think it is good to fix that in this PR

**Comment by rbuj** [comment]
@sc0w 
I think that is fixed now, I applied also the Fedora patch :)

**Comment by lukefromdc** [comment]
Note that 
https://github.com/mate-desktop/mate-system-monitor/pull/151
which depends on this has been merged, so we now need this

**Comment by lukefromdc** [comment]
We should bump version on this to 1.23, as this is a new feature with new dependencies and thus can't be pushed to distros under the same major version number

**Pull request merged**